Src/Plugins/Security/SEC101_011.SquareCredentialsValidator.cs

```diff
@@ -43,5 +60,68 @@ protected override IEnumerable<ValidationResult> IsValidStaticHelper(Dictionary<
 
             return new[] { validationResult };
         }
+
+        protected override ValidationState IsValidDynamicHelper(ref Fingerprint fingerprint,
+                                                                ref string message,
+                                                                Dictionary<string, string> options,
+                                                                ref ResultLevelKind resultLevelKind)
+        {
+            string id = fingerprint.Id;
+            string secret = fingerprint.Secret;
+
+            const string uri = "https://connect.squareup.com/oauth2/token";
+            const string CodeNotFoundMessage = "Authorization code not found";
+            const string CodeForRequest = "123";
+
+            try
+            {
+                HttpClient client = CreateHttpClient();
+
+                var dict = new Dictionary<string, string>();
+                dict.Add("grant_type", "authorization_code");
+                dict.Add("code", CodeForRequest); // Needs some value for call to work.
+                dict.Add("client_id", id);
+                dict.Add("client_secret", secret);
```

Comment:
you can auto-initialize and order by string length

Reply:
Fixed in iteration 3